internal/staging/stdaddress/README.md

```diff
@@ -0,0 +1,195 @@
+   example, `DecodeAddressV0` and `NewAddressPubKeyHashEcdsaSecp256k1V0`.
+
+The first approach is likely suitable for most callers as it is intended to
+support all historical address types as well as allow easily specifying a script
+version from a dynamic variable.  For example, applications involving historical
+data analytics, such as block explorers, should prefer this approach.
+
+However, callers might wish to only support decoding and creating specific
+address versions in which case the second approach is more suitable.  Further,
+when it comes to creating addresses, since not all adddress types will
+necessarily be supported by all script versions, the second approach makes it
+clear from the API exactly which types of addresses are supported for a given
+version at compile time whereas the first approach requires the caller to check
+for errors at run time.
+
+### Generating Payment Scripts and Displaying a Human-Readable Address String
+
+In order to generalize support for the various standard address types, this
+package provides an `Address` interface which is the primary interface
+implemented by all supported address types.
+
+Use the `PaymentScript` method to obtain the script version associated with the
+address along with a script to pay a transaction output to the address and the `Address` method to get the human-readable string encoding of the address.
+
+### Address Use in the Staking System
+
+A distinguishing aspect of the staking system in Decred is that it imposes
+additional constraints on the types of addresses that can be used versus those
+for regular transactions.  This is because scripts in the staking system are
+tightly controlled by consensus to only permit very specific scripts, unlike
+regular transactions which allow scripts composed of any sequence of opcodes
+that evaluate to true.
+
+In order to support this difference, this package provides the `StakeAddress`
+interface which is only implemented by the address types that are supported by
+the staking system.  This allows callers to determine if an address can be used
+in the staking system by type asserting the interface and then making use of the
+relevant methods provided by the interface to interact with the staking system.
+Refer to the code examples to see this in action as well as the API
+documentation of the interface methods for further details.
+
+### Converting Public Key to Public Key Hash Addresses
+
+It is often necessary to share public keys in interactive and manual off-chain
+protocols, such as multi-party signature creation.  Since public keys are valid
+for any network, it is generally less error prone and more convenient to use
+human-readable public key addresses in this case because they ensure the correct
+network is involved as well as help prevent entry errors.
+
+However, whenever dealing with on-chain transactions, public key hash addresses
+should typically be used instead to avoid revealing the public key until
```

Comment:
Should this be `unless absolutely necessary`?

Reply:
Same as previous.